shader/antialiasing/cmaa2/cmaa2_common.glsl

```diff
@@ -0,0 +1,176 @@
+layout(binding = 4) buffer UboWorkingDeferredBlendLocationList {
+  uint g_workingDeferredBlendLocationList[];
+  };
+
+layout(binding = 5) buffer UboWorkingDeferredBlendItemList {
+  uvec2 g_workingDeferredBlendItemList[];
+  };
+
+layout(r32ui, binding = 6) uniform uimage2D g_workingDeferredBlendItemListHeads;
+
+layout(binding = 7) buffer UboWorkingControlBuffer {
+  uint g_workingControlBuffer[];
+  };
+
+#define CmaaSaturate(x) clamp(x, 0.0, 1.0) 
+
+uvec4 UnpackEdges(uint value) {
+  uvec4 ret;
+  ret.x = uint((value & 0x01) != 0);
+  ret.y = uint((value & 0x02) != 0);
+  ret.z = uint((value & 0x04) != 0);
+  ret.w = uint((value & 0x08) != 0);
+  return ret;
+  }
+
+vec4 UnpackEdgesFlt(uint value) {
+  vec4 ret;
+  ret.x = float((value & 0x01) != 0);
+  ret.y = float((value & 0x02) != 0);
+  ret.z = float((value & 0x04) != 0);
+  ret.w = float((value & 0x08) != 0);
+  return ret;
+  }
+
+vec3 LoadSourceColor(ivec2 pixelPos, ivec2 offset, int sampleIndex) {
+  vec3 color = texelFetch(g_inoutColorReadonly, pixelPos + offset, 0).xyz;
+  return color;
+  }
+
+uint Pack_R11G11B10_FLOAT(vec3 rgb) {
+  rgb = min(rgb, uintBitsToFloat(0x477C0000));
+  uint r = ((floatBitsToUint(rgb.x) + 8) >> 4) & 0x000007FF;
+  uint g = ((floatBitsToUint(rgb.y) + 8) << 7) & 0x003FF800;
+  uint b = ((floatBitsToUint(rgb.z) + 16) << 17) & 0xFFC00000;
+  return r | g | b;
+  }
+
+vec3 Unpack_R11G11B10_FLOAT(uint rgb) {
+  float r = uintBitsToFloat((rgb << 4) & 0x7FF0);
+  float g = uintBitsToFloat((rgb >> 7) & 0x7FF0);
+  float b = uintBitsToFloat((rgb >> 17) & 0x7FE0);
+  return vec3(r, g, b);
+  }
+
+uint PackFloat32AsFloat16AndConvertToUint(float arg) {
+  return packHalf2x16(vec2(arg, 0.));
+  }
+
+float UnpackUintAsFloat16AndConvertToFloat32(uint arg) {
+  return unpackHalf2x16(arg).x;
+  }
+
+uint Pack_R11G11B10_E4_FLOAT(vec3 rgb) {
+  rgb = clamp(rgb, 0.0, uintBitsToFloat(0x3FFFFFFF));
+  uint r = ((PackFloat32AsFloat16AndConvertToUint(rgb.r) + 4) >> 3) & 0x000007FF;
+  uint g = ((PackFloat32AsFloat16AndConvertToUint(rgb.g) + 4) << 8) & 0x003FF800;
+  uint b = ((PackFloat32AsFloat16AndConvertToUint(rgb.b) + 8) << 18) & 0xFFC00000;
+  return r | g | b;
+  }
+
+vec3 Unpack_R11G11B10_E4_FLOAT(uint rgb) {
+  float r = UnpackUintAsFloat16AndConvertToFloat32((rgb << 3) & 0x3FF8);
+  float g = UnpackUintAsFloat16AndConvertToFloat32((rgb >> 8) & 0x3FF8);
+  float b = UnpackUintAsFloat16AndConvertToFloat32((rgb >> 18) & 0x3FF0);
+  return vec3(r, g, b);
+  }
+
+vec3 InternalUnpackColor(uint packedColor) {
+#if CMAA2_SUPPORT_HDR_COLOR_RANGE
+  return Unpack_R11G11B10_FLOAT(packedColor);
+#else
+  return Unpack_R11G11B10_E4_FLOAT(packedColor);
+#endif
+  }
+
+uint InternalPackColor(vec3 color) {
+#if CMAA2_SUPPORT_HDR_COLOR_RANGE
+  return Pack_R11G11B10_FLOAT(color);
+#else
+  return Pack_R11G11B10_E4_FLOAT(color);
+#endif
+  }
+
+float LINEAR_to_SRGB(float val) {
+  if(val < 0.0031308) val *= 12.92;
+  else val = 1.055 * pow(abs(val), 1.0 / 2.4) - 0.055;
+  return val;
```

Comment:
OpenGothic uses more simple curve in `srgbEncode`.
Should we use more advance gamma-correction and tonemapping is orthogonal question, for now let's keep code coherent.